great_expectations/experimental/datasources/spark_s3_datasource.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import TYPE_CHECKING, Any, Dict, Optional, Union
+
+import pydantic
+from typing_extensions import Literal
+
+from great_expectations.core.util import S3Url
+from great_expectations.experimental.datasources import _SparkFilePathDatasource
+from great_expectations.experimental.datasources.data_asset.data_connector import (
+    DataConnector,
+    S3DataConnector,
+)
+from great_expectations.experimental.datasources.interfaces import TestConnectionError
+from great_expectations.experimental.datasources.spark_datasource import (
+    SparkDatasourceError,
+)
+from great_expectations.experimental.datasources.spark_file_path_datasource import (
+    CSVAsset,
+)
+
+if TYPE_CHECKING:
+    from botocore.client import BaseClient
+
+    from great_expectations.experimental.datasources.interfaces import (
+        BatchSorter,
+        BatchSortersDefinition,
+    )
+
+
+logger = logging.getLogger(__name__)
+
+
+BOTO3_IMPORTED = False
+try:
+    import boto3
+
+    BOTO3_IMPORTED = True
+except ImportError:
+    pass
+
+
+class SparkS3DatasourceError(SparkDatasourceError):
+    pass
+
+
+class SparkS3Datasource(_SparkFilePathDatasource):
+    # instance attributes
+    type: Literal["spark_s3"] = "spark_s3"
+
+    # S3 specific attributes
+    bucket: str
+    boto3_options: Dict[str, Any] = {}
+
+    _s3_client: Union[BaseClient, None] = pydantic.PrivateAttr(default=None)
+
+    def _get_s3_client(self) -> BaseClient:
+        s3_client: Union[BaseClient, None] = self._s3_client
+        if not s3_client:
+            # Validate that "boto3" libarary was successfully imported and attempt to create "s3_client" handle.
+            if BOTO3_IMPORTED:
+                try:
+                    s3_client = boto3.client("s3", **self.boto3_options)
+                except Exception as e:
+                    # Failure to create "s3_client" is most likely due invalid "boto3_options" dictionary.
+                    raise SparkS3DatasourceError(
+                        f"""Due to exception: "{str(e)}", "s3_client" could not be created."""
```

Comment:
This doesn't need to be a docstring/multiline string.
```suggestion
                        f"Due to exception: "{str(e)}", "s3_client" could not be created."
```

Reply:
@Kilo59 This change will not work, unless we change outer double quotes to single quotes.

Reply:
Didn't notice that. But yes, that is preferable. You can also use single quotes `'` on the inside of the string.

Reply:
I will follow up on your suggestion by using the single quotes for `Python`.  Thanks.